src/big/big_float.cr

```diff
@@ -18,7 +18,37 @@ struct BigFloat < Float
   end
 
   def initialize(num : Number)
-    LibGMP.mpf_init_set_d(out @mpf, num.to_f64)
+    case num
```

Comment:
Does this new arrangement where there is no `Number` overload trigger the bug? If no we should use overloads here.

Reply:
Yes, it is workaround.
Here was comment "XXX fix me when overload will work", but Akzhan removed it.

Reply:
Now will be extracted as issue and link added

Reply:
But does this *specific* code still break the specs. If it does not (and I don't belueve it does) the workaround should be removed.

Reply:
@RX14 this code once breaks the specs in some ways, use case has been extracted to #4897.

We must do this way until #4897 be fixed.

Reply:
You didn't answer my question, does this *specific* code break the specs or not?

Reply:
Just tested, extracted overloads works ok.

But it doesn't OK because I don't know why it sometimes didn't work - as exposed by #4897. 

Compiler has unpredictable behavior to choose which overload will be chosen :-/

Reply:
@RX14 - Moreover, added overloads break specs at all. Again.

  1) BigFloat new new(BigInt)
     Failure/Error: bigfloat_on_bigint_value.should eq(bigfloat_of_integer_value)
       Expected: 123456789012345678901_big_f
            got: 123456789012345667584_big_f
     # spec/std/big/big_float_spec.cr:18
  2) BigFloat new new(BigRational)
     Failure/Error: bigfloat_on_bigrational_value.should eq(BigFloat.new(1) / BigFloat.new(3))
       Expected: 0.333333333333333333333_big_f
            got: 0.33333333333333331483_big_f

Reply:
I hate Crystal compiler overloads for now. They have unpredictable behaviour.

Reply:
@RX14 What can I do with these compiler bugs? I want to rollback proposed overloads due to its erroneous. .

Reply:
by the way - bin/crystal spec/std/big/big_float_spec.cr works ok (failed only std spec)

Reply:
The bug is very clearly related to using `Number` or `Int` overloads (both of which are superclasses of `BigInt`). Simply avoid such generic overloads. Your `case` statement didn't have them, you don't need them.

There is no need to be "scared" of a bug, there is almost 0 chance that the bug will depend on anything but the list of overloads and their order.

Reply:
I have initialize(num : Number) anyway